app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -297,6 +280,25 @@ def index_for_group_type
     end
   end
 
+  private def create_council_role(group, user_id, status)
+    team = group.team
+    return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group.id)
+    if status == "leader"
+      old_leader = Team.find_by(id: team.id).leader
+      if old_leader.present?
+        old_leader.update!(end_date: Date.today)
+      end
+    end
+    already_existing_member = TeamMember.find_by(team_id: team.id, user_id: user_id, end_date: nil)
+    if already_existing_member.present?
```

Comment:
Not really. I think the naming caused a confusion here.

`old_leader`: The person who was having the leader role earlier. Not the person whose role is going to be changed.
`already_existing_member`: This is the `TeamMember` row of the person whose role is going to be changed. Since for the new role (irrespective of whether a leader or not) we need to end the existing role, this is done.

I understand that `already_existing_member` name looks confusing. I've renamed it to `already_existing_row`, but couldn't find a better name. I've also added couple of comments to make it more clear.

Reply:
Okay, thanks for the additional explanation :) Makes total sense now!